tests/unittests/migrations/runner_tests.py

```diff
@@ -0,0 +1,65 @@
+"""
+| Copyright 2017-2025, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+
+import datetime
+import unittest
+from unittest.mock import patch, MagicMock
+
+from fiftyone.migrations import get_datasets_revisions
+
+
+class TestGetDatasetRevisions(unittest.TestCase):
+    @patch("fiftyone.core.odm.get_db_conn")
+    def test_all_fields_present(self, mock_get_db_conn):
+        mock_conn = MagicMock()
+        mock_conn.datasets.find.return_value = [
+            {"name": "dataset1", "version": "v1"},
+            {"name": "dataset2", "version": "v2"},
+        ]
+        mock_get_db_conn.return_value = mock_conn
+
+        result = get_datasets_revisions()
+        expected = {"dataset1": "v1", "dataset2": "v2"}
+        self.assertEqual(result, expected)
+
+    @patch("fiftyone.core.odm.get_db_conn")
+    def test_missing_version_field(self, mock_get_db_conn):
+        mock_conn = MagicMock()
+        mock_conn.datasets.find.return_value = [
+            {"name": "dataset1"},  # version is missing
+            {"name": "dataset2", "version": "v2"},
+        ]
+        mock_get_db_conn.return_value = mock_conn
+
+        result = get_datasets_revisions()
+        expected = {"dataset1": None, "dataset2": "v2"}
+        self.assertEqual(result, expected)
+
+    @patch("fiftyone.core.odm.get_db_conn")
+    def test_empty_collection(self, mock_get_db_conn):
+        mock_conn = MagicMock()
+        mock_conn.datasets.find.return_value = []
+        mock_get_db_conn.return_value = mock_conn
+
+        result = get_datasets_revisions()
+        self.assertEqual(result, {})
+
+    def test_missing_name(self):
```

Comment:
This test scares me. There's no automatic mechanism to cleanup bad `datasets` docs so if this `finally` block ever fails, eg due to a ctrl + c, then there is forever a bad doc in the user's `datasets` collection.

Reply:
tbh I also didn't like having to write an invalid dataset to a real db. I'm happy to remove it.